control_plane/storcon_cli/src/main.rs

```diff
@@ -0,0 +1,643 @@
+    TenantLocateResponse, TenantShardMigrateRequest, TenantShardMigrateResponse,
+};
+
+#[derive(Subcommand, Debug)]
+enum Command {
+    NodeRegister {
+        #[arg(long)]
+        node_id: NodeId,
+
+        #[arg(long)]
+        listen_pg_addr: String,
+        #[arg(long)]
+        listen_pg_port: u16,
+
+        #[arg(long)]
+        listen_http_addr: String,
+        #[arg(long)]
+        listen_http_port: u16,
+    },
+    NodeConfigure {
+        #[arg(long)]
+        node_id: NodeId,
+
+        #[arg(long)]
+        availability: Option<NodeAvailabilityWrapper>,
+        #[arg(long)]
+        scheduling: Option<NodeSchedulingPolicy>,
+    },
+    TenantPolicy {
+        #[arg(long)]
+        tenant_id: TenantId,
+        #[arg(long)]
+        placement: Option<PlacementPolicy>,
+        #[arg(long)]
+        scheduling: Option<ShardSchedulingPolicy>,
+    },
+    Nodes {},
+    Tenants {},
+    TenantCreate {
+        #[arg(long)]
+        tenant_id: TenantId,
+    },
+    TenantDelete {
+        #[arg(long)]
+        tenant_id: TenantId,
+    },
+    TenantShardSplit {
+        #[arg(long)]
+        tenant_id: TenantId,
+        #[arg(long)]
+        shard_count: u8,
+        #[arg(long)]
+        stripe_size: Option<u32>,
+    },
+    TenantShardMigrate {
+        #[arg(long)]
+        tenant_shard_id: TenantShardId,
+        #[arg(long)]
+        node: NodeId,
+    },
+    TenantConfig {
+        #[arg(long)]
+        tenant_id: TenantId,
+        #[arg(long)]
+        config: String,
+    },
+    ServiceRegister {
+        #[arg(long)]
+        http_host: String,
+        #[arg(long)]
+        http_port: u16,
+        #[arg(long)]
+        region_id: String,
+        #[arg(long)]
+        availability_zone_id: String,
+        #[arg(long)]
+        version: u64,
+        // {
+        //   "version": 1,
+        //   "host": "${HOST}",
+        //   "port": 6400,
+        //   "region_id": "{{ console_region_id }}",
+        //   "instance_id": "${INSTANCE_ID}",
+        //   "http_host": "${HOST}",
+        //   "http_port": 9898,
+        //   "active": false,
+        //   "availability_zone_id": "${AZ_ID}",
+        //   "disk_size": ${DISK_SIZE},
+        //   "instance_type": "${INSTANCE_TYPE}",
+        //   "register_reason" : "New pageserver"
+        // }
+    },
+    TenantOnboard {
+        #[arg(long)]
+        tenant_id: TenantId,
+        #[arg(long)]
+        target_pageserver_id: NodeId,
+    },
+    TenantScatter {
+        #[arg(long)]
+        tenant_id: TenantId,
+    },
+    TenantDescribe {
+        #[arg(long)]
+        tenant_id: TenantId,
+    },
+}
+
+/// Request format for control plane POST /management/api/v2/pageservers
+///
+/// This is how physical pageservers register, but in this context it is how we
+/// register the storage controller with the control plane, as a "virtual pageserver"
+#[derive(Serialize, Deserialize, Debug)]
+struct ServiceRegisterRequest {
```

Comment:
Why is this (and `ServiceRegisterResponse`) not part of `libs/pageserver_api/src/controller_api.rs` ?

Reply:
It's part of the cloud control plane API, rather than the controller API (but this is gone now, since I removed service-register)